pkg/engine/engine.go

```diff
@@ -132,193 +116,280 @@ func (e *QueryEngine) Execute(ctx context.Context, params logql.Params) (logqlmo
 	))
 	defer span.End()
 
-	var (
-		startTime = time.Now()
+	if err := e.tryInit(); err != nil {
```

Comment:
What is the reason why the engine is initialized lazily upon first execution?

Reply:
this is gone with the change back to a constructor